scripts/common/common.sh

```diff
@@ -1151,3 +1156,307 @@ function cmd_retry() {
     done
     return 0
 }
+
+# common_upgrade_step_versions returns a list of upgrade steps that need to be performed, based on
+# the supplied step versions, for use by other functions.
+# e.g. "1.5.12\n1.6.11\n1.7.11"
+function common_upgrade_step_versions() {
+    declare -a _step_versions=("${!1}")
+    local from_version=$2
+    local to_version=$3
+
+    # check that both are major version 1
+    if  [ "$(common_upgrade_major_minor_to_major "$from_version")" != "1" ] || \
+        [ "$(common_upgrade_major_minor_to_major "$to_version")" != "1" ] ; then
+        bail "Upgrade from $from_version to $to_version is not supported."
+    fi
```

Comment:
Why do we restrict this to major version 1 ? Wouldn't it suffice for both to be the same version instead ? Like both being in major version 2.

Reply:
fixed to compare majors but no longer restrict